Assets/Scripts/Bullet/BulletController.cs

```diff
@@ -30,9 +30,10 @@ public void Fire(Vector3 eulerAngle, float shootSpeed)
         private void OnCollisionEnter(Collision collision)
         {
             ShowShellExplosion(collision.contacts[0].point);
-            if (collision.gameObject.GetComponent<TankController>() != null)
+            IDamageable damageable = collision.gameObject.GetComponent<IDamageable>();
+            if (damageable != null)
             {
-                DamageTank(collision.gameObject.GetComponent<TankController>());
+                DamageTank(damageable);
```

Comment:
Change DamageTank function as this bullet can damage anything that is damageable